src/System.Net.Security/src/System/Net/NTAuthentication.cs

```diff
@@ -58,10 +57,10 @@ internal string AssociatedName
             {
                 if (!(IsValidContext && IsCompleted))
                 {
-                    throw new Win32Exception((int)Interop.SecurityStatus.InvalidHandle);
+                    throw new Exception(SR.net_auth_noauth);
```

Comment:
Why is this changing from Win32Exception to Exception?  I realize the naming is poor, but it is actually supported on Unix, and it has the key features of both storing the error code and mapping it to a string supplied by the OS.  Changing it to be Exception as is done here would seem to lose both of those things.